jupyter_server/serverapp.py

```diff
@@ -1427,6 +1434,12 @@ def _default_allow_remote(self) -> bool:
                         """,
     )
 
+    allow_custom_env_variables = Bool(
+        False,
+        config=True,
+        help="""Allow to use custom env variables""",
+    )
+
```

Comment:
I think this trait name and description is too vague. 

Since this is specifically about passing custom environment variables to the kernel, I think we need to clarify in this trait and help message that we mean kernel env vars here.